django/contrib/gis/gdal/raster/source.py

```diff
@@ -173,6 +213,25 @@ def _flush(self):
         capi.flush_ds(self._ptr)
 
     @property
+    def vsi_buffer(self):
+        if not self.is_vsi_based:
+            return None
```

Comment:
Added test for it [here](https://github.com/django/django/pull/8634/files#diff-469b4a6b934be489c22f3cf20c6e0375R218).